### PR TITLE
kv/storage: disambiguate test

### PR DESCRIPTION
**Pull request by tamird**
This test was testing `SplitTrigger` in `kv`, with a comment that
indicated it was testing permissions. This moves the `SplitTrigger`
test to `storage` and adds a test for `InternalCommitTrigger`
permissions in `kv`.

@tschottdorf this is what made cherry-picking your client change necessary; the error that was returned was about malformed splits (because `verifyRequest` in `kv` wasn't actually checking `EndTrasaction` in batches). When I fixed that in #2351, this test started failing because _that_ error wasn't coming back until your fix was in.

### CODE REVIEW

File: kv/db_test.go [hunk 1]
```diff
@@ -169,7 +173,7 @@ func TestKVDBInternalMethods(t *testing.T) {
 		err := db.Run(b)
 		if err == nil {
 			t.Errorf("%d: unexpected success calling %s", i, args.Method())
-		} else if !strings.Contains(err.Error(), "couldn't find method") {
+		} else if !(strings.Contains(err.Error(), "couldn't find method") || strings.Contains(err.Error(), "contains commit trigger")) {
```
Comment (tschottdorf) [inline kv/db_test.go]: not `testutils.IsError`?

File: kv/db_test.go [hunk 2]
```diff
@@ -181,45 +185,12 @@ func TestKVDBInternalMethods(t *testing.T) {
 
 		if err := db.Run(b); err == nil {
 			t.Errorf("%d: unexpected success calling %s", i, args.Method())
-		} else if !strings.Contains(err.Error(), "contains an internal request") {
-			t.Errorf("%d: expected disallowed method %s; got %s", i, args.Method(), err)
+		} else if !(strings.Contains(err.Error(), "contains an internal request") || strings.Contains(err.Error(), "contains commit trigger")) {
```
Comment (tschottdorf) [inline kv/db_test.go]: ditto

**Comment by tschottdorf** [comment]
:+1: LGTM

File: kv/db_test.go (hunk 2 above)
Comment (tamird) [inline kv/db_test.go]: done

File: kv/db_test.go (hunk 1 above)
Comment (tamird) [inline kv/db_test.go]: done

**Pull request merged**